File: src/types/directory.ts
export interface ProjectDirectory {
  id: string;
  project_id: string;
  path: string;
  is_primary: boolean;
  status: "ok" | "missing" | "inaccessible" | "error";
  last_scanned_at: string | null;
  created_at: string;
}

export interface ProjectFile {
  id: string;
  directory_id: string;
  relative_path: string;
  size_bytes: number;
  mtime: string;
  content_hash: string | null;
  content_kind: "none" | "text" | "pdf" | "office" | "other";
  extract_status: "ok" | "skipped_too_large" | "unsupported" | "error";
  indexed_at: string;
}

export interface CloudRule {
  id: string;
  directory_id: string;
  scope: "directory" | "file";
  relative_path: string;
  allow: boolean;
}

export interface ProjectContext {
  project_id: string;
  cached_context: string | null;
  context_hash: string | null;
  inventory_hash: string | null;
  allow_cloud_context: boolean;
  generated_at: string | null;
}

export interface RescanOutcome {
  status: string;
  regenerated: boolean;
  file_count: number;
}

File: src-tauri/src/models/directory.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectDirectory {
    pub id: String,
    pub project_id: String,
    pub path: String,
    pub is_primary: bool,
    pub status: String, // 'ok' | 'missing' | 'inaccessible' | 'error'
    pub last_scanned_at: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectFile {
    pub id: String,
    pub directory_id: String,
    pub relative_path: String,
    pub size_bytes: i64,
    pub mtime: String,
    pub content_hash: Option<String>,
    pub content_kind: String, // 'none' | 'text' | 'pdf' | 'office' | 'other'
    pub extract_status: String, // 'ok' | 'skipped_too_large' | 'unsupported' | 'error'
    pub indexed_at: String,
}

/// スキャン結果の1ファイル分（DBに入る前の形）。
#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct ProjectFileEntry {
    pub relative_path: String,
    pub size_bytes: i64,
    pub mtime: String,
    pub content_hash: Option<String>,
    pub content_kind: String,
    pub extract_status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CloudRule {
    pub id: String,
    pub directory_id: String,
    pub scope: String, // 'directory' | 'file'
    pub relative_path: String,
    pub allow: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectContext {
    pub project_id: String,
    pub cached_context: Option<String>,
    pub context_hash: Option<String>,
    pub inventory_hash: Option<String>,
    pub allow_cloud_context: bool,
    pub generated_at: Option<String>,
}
